### ISSUE TITLE
加载已训练好的模型时，会报错IndexError: Can not load the data without registration !

### ISSUE THREAD

**Issue by SanaMoko** [RUCAIBox/RecBole#2113]
**描述这个 bug**

加载已训练好的模型时，会报错IndexError: Can not load the data without registration !

**如何复现**
复现这个 bug 的步骤：
python run_model.py --model=BPR --dataset=Book-Crossing
import argparse
from logging import getLogger
import sys

from utils import get_model
from config import Config
from recbole.utils import init_logger, init_seed, set_color
from recbole.data import create_dataset, data_preparation
from recbole.utils import init_logger, init_seed, set_color

from trainer import Trainer
from recbole.quick_start.quick_start import load_data_and_model
def run_model(model_name, dataset_name, fairness_type):
    props = [
        "props/overall.yaml",
        f"props/{dataset_name}.yaml",
        f"props/{model_name}.yaml",
    ]
    print(props)
    model_class = get_model(model_name)

    # 配置文件初始化
    config = Config(
        model=model_class,
        dataset=dataset_name,
        config_file_list=props,
    )
    init_seed(config["seed"], config["reproducibility"])# 确保全局一致性

    #初始化logger
    init_logger(config)
    logger = getLogger()
    logger.info(sys.argv)
    logger.info(config)

    # 数据集过滤
    dataset= create_dataset(config)
    logger.info(dataset)

    # dataset splitting
    train_data, valid_data, test_data = data_preparation(config, dataset)
    
    # 模型加载 初始化
    init_seed(config["seed"]+config["local_rank"],config["reproducibility"]) #确保模型初始化一致
    model = model_class(config,train_data._dataset).to(config["device"])
    logger.info(model)

    # 加载训练器和初始化
    # trainer = Trainer(config, model)

    # 模型训练
    # best_valid_score,best_valid_result = trainer.fit(
    #     train_data,valid_data,saved = True,show_progress=config["show_progress"]
    # )
    config, model_rec, dataset, train_data, valid_data, test_data = load_data_and_model(
        model_file='Recbole_ours/saved/BPR-Nov-18-2024_13-12-26.pth',
    )
    trainer = Trainer(config, model_rec)

    test_result = trainer.evaluate(
        test_data,load_best_model=False,model_file='Recbole_ours/saved/BPR-Nov-18-2024_13-12-26.pth',show_progress=config["show_progress"]
    )
    logger.info(set_color("test_result","yellow")+f":{test_result}")

    return (
        model_name,
        dataset_name,
        {
            "test_result": test_result,
        },
    )
    

**预期**
希望能够给出一个加载已训练好的模型进行测试的示例。
